File: examples/python/meta.py
#!/usr/bin/env python3
"""读取实例公开信息。依赖: pip install requests"""
import json
import os
import sys

import requests

INSTANCE = os.environ.get("INSTANCE", "https://example.com").rstrip("/")


def main() -> None:
    r = requests.post(
        f"{INSTANCE}/api/meta",
        json={"detail": True},
        headers={"Accept": "application/json"},
        timeout=30,
    )
    if not r.ok:
        print(r.status_code, r.text, file=sys.stderr)
        sys.exit(1)
    meta = r.json()
    print(json.dumps({
        "name": meta.get("name"),
        "version": meta.get("version"),
        "uri": meta.get("uri"),
    }, ensure_ascii=False, indent=2))


if __name__ == "__main__":
    main()

File: examples/rust/src/meta.rs
//! 读取实例公开信息
//! 运行: cd examples/rust && cargo run --bin meta
//! 环境变量: INSTANCE=https://example.com

use serde_json::json;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let instance = std::env::var("INSTANCE").unwrap_or_else(|_| "https://example.com".into());
    let instance = instance.trim_end_matches('/');
    let client = reqwest::blocking::Client::new();
    let v: serde_json::Value = client
        .post(format!("{instance}/api/meta"))
        .json(&json!({ "detail": true }))
        .send()?
        .error_for_status()?
        .json()?;
    println!(
        "{}",
        serde_json::json!({
            "name": v.get("name"),
            "version": v.get("version"),
            "uri": v.get("uri"),
        })
    );
    Ok(())
}
